apps/dashboard/app/models/batch_connect/session.rb

```diff
@@ -208,7 +208,11 @@ def save(app:, context:, format: nil)
       self.view       = app.session_view
       self.created_at = Time.now.to_i
 
-      submit_script = app.submit_opts(context, fmt: format) # could raise an exception
+      # add staged_root to rendering context
+      bind = context.get_binding
+      bind.local_variable_set(:staged_root, self.staged_root)
+
+      submit_script = app.submit_opts(bind, context, fmt: format) # could raise an exception
```

Comment:
Something about passing the binding here doesn't feel right. Like OodApp#submit_opts shouldn't know that much. Maybe we should just add `staged_root` as a key/value pair to the context hash?

Reply:
@ericfranz any thoughts on this? I've gone over this a few times and this seems to be the smallest amount of code to add.  

`SessionContext` seems to be read only, so another route I was maybe thinking was create an Attribute to that is this `staged_root` and add it to this SesssionContext, but that would increase it's interface a bit.

Reply:
I think https://github.com/OSC/ondemand/pull/948 is the correct solution. Though it would have taken too long to explain it so I just wrote it up. Didn't have time to test or anything though.

It's a partial solution to https://github.com/OSC/ondemand/issues/605 as well but needs more work.

You are right, submit_opts should be responsible for its own rendering (and thus what the binding is). Also calling staged_root has a side effect unfortunately, and submit_opts is supposed to be able to raise an exception (for validation purposes for example).